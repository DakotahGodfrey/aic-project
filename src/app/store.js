import { configureStore } from '@reduxjs/toolkit';
import homeReducer from 'app/slices/homeSlice';
import galleryReducer from 'app/slices/gallerySlice';
import artistsReducer from 'app/slices/artistsSlice';
import artworksReducer from 'app/slices/artworkSlice';
import artistReducer from 'app/slices/artistSlice';

export const store = configureStore({
  reducer: {
    home: homeReducer,
    gallery: galleryReducer,
    artists: artistsReducer,
    artwork: artworksReducer,
    artist: artistReducer,
  },
});
