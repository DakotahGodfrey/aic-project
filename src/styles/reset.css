*,
*::before,
*::after {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
}
html {
  font-size: 10px;
}
ol,
ul {
  list-style: none;
}
a {
  text-decoration: none;
}
table {
  border-collapse: collapse;
  border-spacing: 0;
}
::-webkit-scrollbar {
  width: 0; /* Remove scrollbar space */
  background: transparent; /* Optional: just make scrollbar invisible */
}
