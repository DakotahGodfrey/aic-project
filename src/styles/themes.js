const defaultTheme = {
  textColor: '#222',
  secondaryColor: '#fff',
  navBG: '#111',
  lightColor: '#eee',
  brandColor: '#B50938',
  brandColorFaded: 'rgba(181,9,56, 0.7)',
  serifFonts: `Georgia, "Times New Roman", Times, serif`,
  sansSerifFonts: `Helvetica, sans-serif`,
  mutedText: '#666',
  linkHover: '#22959c',
  lightGrey: '#F3F3F3',
};
const darkTheme = {
  textColor: '#eee',
  navBG: '#111',
  secondaryColor: '#151515',
  lightColor: '#eee',
  brandColor: '#B50938',
  brandColorFaded: 'rgba(181,9,56, 0.7)',
  serifFonts: `Georgia, "Times New Roman", Times, serif`,
  sansSerifFonts: `Helvetica, sans-serif`,
  mutedText: '#ddd',
  linkHover: '#22959c',
  lightGrey: '#F3F3F3',
};

const getFromTheme = (propName) => (props) => props.theme[propName];
export { defaultTheme, darkTheme, getFromTheme };
