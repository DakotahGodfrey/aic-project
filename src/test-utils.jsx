import * as React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';

function render(ui, { theme, store, ...options } = {}) {
  const Wrapper = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <Router>{children}</Router>
      </ThemeProvider>
    </Provider>
  );
  return rtlRender(ui, { wrapper: Wrapper, ...options });
}

export * from '@testing-library/react';
// override React Testing Library's render with our own
export { render };
