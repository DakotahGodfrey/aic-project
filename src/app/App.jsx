import React, { useEffect, useState } from 'react';
import { ThemeProvider } from 'styled-components';
import { darkTheme, defaultTheme } from 'styles/themes';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  useLocation,
} from 'react-router-dom';
import SiteNav from 'components/layout/navigation/SiteNav';
import { siteLinks } from 'utils/data';
import Home from 'components/screens/Home';
import Gallery from 'components/screens/Gallery';
import Artists from 'components/screens/Artists';
import Artwork from 'components/screens/Artwork';
import Artist from 'components/screens/Artist';
import AboutApp from 'components/screens/AboutApp';
import NotFound from 'components/screens/NotFound';
import ScrollUpButton from 'components/layout/buttons/ScrollUpButton';

function ScrollToTop() {
  const { pathname } = useLocation();

  useEffect(() => {
    if (window instanceof Window) {
      window.scrollTo(0, 0);
    }
  }, [pathname]);

  return null;
}

const App = () => {
  const [theme, setTheme] = useState('default');

  const handleThemeToggle = () => {
    if (theme === 'default') {
      setTheme('darkTheme');
    } else {
      setTheme('default');
    }
    console.log(theme);
  };
  return (
    <ThemeProvider theme={theme === 'default' ? defaultTheme : darkTheme}>
      <Router>
        <div
          id='app'
          style={
            theme === 'default'
              ? { backgroundColor: '#eee' }
              : { backgroundColor: '#111' }
          }
        >
          <ScrollToTop />
          <SiteNav
            links={siteLinks}
            toggleTheme={handleThemeToggle}
            theme={theme}
          />
          <Switch>
            <Route exact path={'/'} component={Home} />
            <Route exact path={'/gallery'} component={Gallery} />
            <Route exact path={'/artists'} component={Artists} />
            <Route exact path={'/gallery/artwork/:id'} component={Artwork} />
            <Route exact path={'/artists/:id'} component={Artist} />
            <Route exact path={'/about'} component={AboutApp} />
            <Route component={NotFound} />
          </Switch>
          <ScrollUpButton />
        </div>
      </Router>
    </ThemeProvider>
  );
};

export default App;
