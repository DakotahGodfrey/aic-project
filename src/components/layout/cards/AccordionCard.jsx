import styled from 'styled-components';
import { getFromTheme } from 'styles/themes';
import React from 'react';

const AccordionWrapper = styled.div`
  font-size: 1.7rem;
  button {
    font-size: inherit;
    font-weight: 700;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 0.5rem;
    background-color: transparent;
    color: ${getFromTheme('textColor')};
    width: 100%;
    margin: 2rem 0 0 0;
    &:hover {
      color: ${getFromTheme('brandColor')};
    }
  }
  .body {
    padding: 0 2.2rem;
    overflow-y: scroll;
    overflow-x: hidden;
    color: ${getFromTheme('textColor')};
    border-left: 1px solid ${getFromTheme('mutedText')};
    max-height: ${(props) => (props.isActive ? '120rem' : 0)};
    list-style-type: disc;
    li {
      margin-bottom: 1rem;
    }
  }
  p {
    padding: 1rem;
  }
`;

const AccordionCard = ({ heading, body, isArray }) => {
  const [isActive, setIsActive] = React.useState(false);
  const handleClick = () => {
    setIsActive(!isActive);
  };
  return (
    <AccordionWrapper isActive={isActive} handleClick={handleClick}>
      <button onClick={handleClick} aria-expanded={isActive}>
        {heading}
        <span className='material-icons'>{isActive ? 'remove' : 'add'}</span>
      </button>

      {isArray ? (
        <ul className='body'>
          {body
            .filter((str) => /\S/.test(str))
            .map((str, index) => (
              <li key={index}>{str}</li>
            ))}
        </ul>
      ) : (
        <div className='body' role='dialog'>
          <p>{body}</p>
        </div>
      )}
    </AccordionWrapper>
  );
};

export default AccordionCard;
