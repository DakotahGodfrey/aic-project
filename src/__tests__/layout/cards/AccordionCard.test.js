import { store } from 'app/store';
import AccordionCard from 'components/layout/cards/AccordionCard';
import { defaultTheme } from 'styles/themes';
import { render, screen, fireEvent } from 'test-utils';
import React from 'react';

describe('AccordionCard', () => {
  it('should render', () => {
    const { container } = render(<AccordionCard />, {
      theme: defaultTheme,
      store: store,
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders a button with text equal to the heading prop', () => {
    render(<AccordionCard heading='test' />, {
      theme: defaultTheme,
      store: store,
    });
    expect(screen.getByRole('button')).toHaveTextContent(/test/);
  });

  it('when the button is clicked it toggles the text content', () => {
    render(<AccordionCard heading='test' />, {
      theme: defaultTheme,
      store: store,
    });
    const button = screen.getByRole('button');
    expect(button).toHaveTextContent(/add/);
    fireEvent.click(button);
    expect(button).toHaveTextContent(/remove/);
  });

  it('when the button is clicked toggles the max-height of the element with className body and textcontent equal to the body prop', () => {
    render(<AccordionCard heading='test' body='body' />, {
      theme: defaultTheme,
      store: store,
    });
    const button = screen.getByRole('button');
    expect(document.querySelector('.body')).toHaveStyle('max-height: 0');
    fireEvent.click(button);
    expect(document.querySelector('.body')).toHaveStyle('max-height: 120rem');
  });

  it('renders a list if the isArray prop is true', () => {
    render(<AccordionCard isArray body={['test', 'test', 'test']} />, {
      theme: defaultTheme,
      store: store,
    });
    expect(screen.getByRole('list')).toBeTruthy();
  });

  it('renders a list item for eachitem in the body prop if the isArray prop is true', () => {
    render(<AccordionCard isArray body={['test', 'test', 'test']} />, {
      theme: defaultTheme,
      store: store,
    });
    expect(screen.getAllByRole('listitem')).toHaveLength(3);
  });

  it('renders a dialog with text content equal to the body prop the isArray prop is false', () => {
    render(<AccordionCard body='test' />, {
      theme: defaultTheme,
      store: store,
    });
    expect(screen.getByRole('dialog')).toHaveTextContent(/test/);
  });
});
