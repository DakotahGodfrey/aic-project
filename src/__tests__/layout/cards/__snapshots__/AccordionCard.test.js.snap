// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionCard should render 1`] = `
<div
  class="sc-bdnxRM cwgERM"
>
  <button
    aria-expanded="false"
  >
    <span
      class="material-icons"
    >
      add
    </span>
  </button>
  <div
    class="body"
    role="dialog"
  >
    <p />
  </div>
</div>
`;
